sopht_simulator/immersed_body/cosserat_rod/cosserat_rod_flow_interaction.py

```diff
@@ -58,3 +61,9 @@ def __init__(
             num_threads,
             start_time,
         )
+
+    def update_rod_element_position(self):
+        self.cosserat_rod.element_position[...] = 0.5 * (
+            self.cosserat_rod.position_collection[..., 1:]
+            + self.cosserat_rod.position_collection[..., :-1]
+        )
```

Comment:
Conceptually this is only a part of the rod visualization and has nothing to do with the flow-rod interactor. I would suggest removing this from here, and rather having a `CosseratRodIO` class derived from `IO` in `backend`, which then specializes in the `.save()` function, calling this update inside there.

Reply:
That's probably a good idea. Let me give this a shot then.

Reply:
@bhosale2 please see if the updated implementation in [1ad159a](https://github.com/SophT-Team/SophT-Simulator/pull/70/commits/1ad159a0deb7c2e448e9b3a74928be13246e5037) works.